test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,386 @@
+   modules in separate files by naming them on the command line, utilizing the
+   -M flag, etc. (see the man page for exact details).
+ */
+module modToUse {
+
+  /* In this case, foo is a public module-level variable that is defined within
+     the module modToUse */
+  var foo = 12;
+
+  /* As is bar. */
+  var bar: int = 2;
+
+  /* A symbol can be declared "private" - this means that only code defined
+     within the same scope as the definition of this symbol can access it.
+
+     Here, hiddenFoo is a private global variable, which is only accessible by
+     symbols defined within modToUse
+  */
+  private var hiddenFoo = false;
+
+
+  /* baz is a public function which is defined within modToUse */
+  proc baz (x, y) {
+    return x * (x + y);
+  }
+
+  /* hiddenBaz is a private function, which is also only accessible by symbols
+     defined within modToUse.
+  */
+  private proc hiddenBaz(a) {
+    writeln(a);
+    return a + 3;
+  }
+
+  // In the current implementation, private cannot be applied to type
+  // definitions; type aliases, and declarations of enums, records, and
+  // classes cannot be declared private.  Private also cannot be applied to
+  // fields or methods yet.
+
+}
+
+module AnotherModule {
+  var a = false;
+}
+
+module ThirdModule {
+  var b = -13.0;
+}
+
+
+module Conflict {
+  /* This variable shares a name with a symbol in modToUse. */
+  var bar = 5;
+
+  var other = 5.0 + 3i;
+
+  var another = false;
+}
+
+
+module DifferentArguments {
+  /* This function shares a name with a function in modToUse, but requires
+     different arguments */
+  proc baz(x) {
+    return x - 2;
+  }
+}
+
+module MainModule {
+  proc main() {
+    writeln("Access from outside a module");
+
+    /* If a module is not the main module for a program, it is desirable for its
+       contents to be accessible to external modules.  There are several
+       strategies for accomplishing this:
+
+       First, a symbol can be referenced explicitly - this is done using the
+       module name and a separating '.' as a prefix to the name of the symbol
+       desired.
+    */
+    var thriceFoo = 3 * modToUse.foo; // should be 36
+    writeln(thriceFoo);
+
+
+    {
+      /* If many of the module's symbols are desired, or the same symbol is
+         desired multiple times, then it is best to utilize what is known as a
+         "use statement".
+      */
+      use modToUse;
+
+      /* Use statements can be inserted at any lexical scope that contains
+         executable code.
+
+         A use statement makes all of the module's visible symbols available to
+         the scope which contains the use statement.  These symbols may then be
+         accessed without the module name prefix.
+
+         In this case, bazBarFoo should store the result of calling modToUse.baz
+         on modToUse.bar and modToUse.foo, which is in this case '28'.
+      */
+      var bazBarFoo = baz(bar, foo);
+      writeln(bazBarFoo);
+    }
```

Comment:
What do you think about throwing a comment just before this curly bracket saying "Since 'use' statements only affect their containing scope, when we leave a scope like this, we revert to requiring fully-qualified names" or something like that.  This would set up your next comment (without requiring the reader to notice that the curly bracket we just left was the one containing the 'use').